src/dotnet/ToolPackage/PackageId.cs

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using Microsoft.DotNet.InternalAbstractions;
+
+namespace Microsoft.DotNet.ToolPackage
+{
+    internal struct PackageId : IEquatable<PackageId>, IComparable<PackageId>
+    {
+        private string _id;
+
+        public PackageId(string id)
+        {
+            _id = id?.ToLowerInvariant() ?? throw new ArgumentNullException(nameof(id));
+        }
+
+        public bool Equals(PackageId other)
+        {
+            return _id == other._id;
+        }
+
+        public int CompareTo(PackageId other)
+        {
+            return _id.CompareTo(other._id);
```

Comment:
Are we ordering these somewhere? For what purpose?

This should probably be `string.Compare(_id, other._id, StringComparison.Ordinal);`

Reply:
The `list tool` command displays the packages in order based on id.

Reply:
Will fix.